internal/controller/storageclaim_controller.go

```diff
@@ -392,7 +415,13 @@ func (r *StorageClaimReconciler) reconcilePhases() (reconcile.Result, error) {
 					storageClass = r.getCephRBDStorageClass(data)
 				}
 				utils.AddAnnotation(storageClass, storageClaimAnnotation, r.storageClaim.Name)
-				err = r.createOrReplaceStorageClass(storageClass)
+				err = r.createOrReplace(storageClass, func() error {
+					if storageClass.Parameters == nil {
+						storageClass.Parameters = map[string]string{}
+					}
+					storageClass.Parameters = data
```

Comment:
We don't need to set to map[string]string{} if we are setting to be equal to data in next line